distribution/graal/oal-rt-graal-native/src/main/java/org/apache/skywalking/oal/rt/OALRuntime.java

```diff
@@ -0,0 +1,673 @@
+import javassist.bytecode.ClassFile;
+import javassist.bytecode.ConstPool;
+import javassist.bytecode.SignatureAttribute;
+import javassist.bytecode.annotation.Annotation;
+import javassist.bytecode.annotation.ClassMemberValue;
+import javassist.bytecode.annotation.IntegerMemberValue;
+import javassist.bytecode.annotation.StringMemberValue;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.JavaVersion;
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.skywalking.oal.rt.output.AllDispatcherContext;
+import org.apache.skywalking.oal.rt.output.DispatcherContext;
+import org.apache.skywalking.oal.rt.parser.AnalysisResult;
+import org.apache.skywalking.oal.rt.parser.OALScripts;
+import org.apache.skywalking.oal.rt.parser.ScriptParser;
+import org.apache.skywalking.oal.rt.parser.SourceColumn;
+import org.apache.skywalking.oap.server.core.analysis.DisableRegister;
+import org.apache.skywalking.oap.server.core.analysis.DispatcherDetectorListener;
+import org.apache.skywalking.oap.server.core.analysis.SourceDispatcher;
+import org.apache.skywalking.oap.server.core.analysis.Stream;
+import org.apache.skywalking.oap.server.core.analysis.StreamAnnotationListener;
+import org.apache.skywalking.oap.server.core.oal.rt.OALCompileException;
+import org.apache.skywalking.oap.server.core.oal.rt.OALDefine;
+import org.apache.skywalking.oap.server.core.oal.rt.OALEngine;
+import org.apache.skywalking.oap.server.core.source.oal.rt.dispatcher.DispatcherClassPackageHolder;
+import org.apache.skywalking.oap.server.core.source.oal.rt.metrics.MetricClassPackageHolder;
+import org.apache.skywalking.oap.server.core.source.oal.rt.metrics.builder.MetricBuilderClassPackageHolder;
+import org.apache.skywalking.oap.server.core.storage.StorageBuilderFactory;
+import org.apache.skywalking.oap.server.core.storage.StorageException;
+import org.apache.skywalking.oap.server.core.storage.annotation.BanyanDB;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * The old logic of this class used runtime class generation, which is not supported by native-image.
+ * Therefore, in the new logic, class files are generated during the compilation process, see (@link org.apache.skywalking.graal.Generator) and loaded at runtime.
+ */
+@Slf4j
+public class OALRuntime implements OALEngine {
```

Comment:
For the implementation, could you share the workflow about when OAL runtime would generate the classes, and package into the jar files, then when how to use them in the runtime?

This part should be important for this PR. We should document this in the comment as well.

Also, this class seems still mixed the existing OAL runtime logic and part of changes, let's discuss about how to narrow down the case and focus the required change, polish the codes.

Reply:
At build time, `skywalking-generate-plugin` call `OALRuntime#generateOALClassFiles`(this method added by me). Based on what you said, I think this part should depart from the OALRuntime.

At runtime, `OALRuntime#start` will be called, and it should be substituted by load classes which we generate before.(old logic is generate class at runtime).

I will try departing `generateOALClassFIles` later, I previously placed it in `OALRuntime` to reuse the existing API, but it seems indeed unnecessary, and indeed, the logic has become less clear.

Reply:
> I will try departing generateOALClassFIles later, I previously placed it in OALRuntime to reuse the existing API, but it seems indeed unnecessary, and indeed, the logic has become less clear.

We could and should reuse the logic, but I don't want to copy the existing codes in the patch codes. You could consider this as a guidance principle for all my given comments.